samples/go/csv/csv-import/perf_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/attic-labs/noms/go/datas"
+	"github.com/attic-labs/noms/go/dataset"
+	"github.com/attic-labs/noms/go/hash"
+	"github.com/attic-labs/noms/go/perf/suite"
+	"github.com/attic-labs/noms/go/types"
+	humanize "github.com/dustin/go-humanize"
+)
+
+// CSV perf suites require the testdata directory to be checked out at $GOPATH/src/github.com/attic-labs/testdata (i.e. ../testdata relative to the noms directory).
+
+// TODO: Add ny-vehicle-registrations test when CSV importing is faster (testdata/ny-vehicle-registrations/20150218.*).
+
+type perfSuite struct {
+	suite.PerfSuite
+	server      *datas.RemoteDatabaseServer
+	sfcBlobHash hash.Hash
+}
+
+func (s *perfSuite) Test01ImportSfCrimeBlobFromTestdata() {
+	assert := s.NewAssert()
+
+	var raw []io.Reader
+
+	s.Pause(func() {
+		// The raw data is split into a bunch of files foo.a, foo.b, etc.
+		glob, err := filepath.Glob(path.Join(s.AtticLabs, "testdata", "sf-crime", "2016-07-28.*"))
+		assert.NoError(err)
+
+		raw = make([]io.Reader, len(glob))
+		for i, m := range glob {
+			r, err := os.Open(m)
+			assert.NoError(err)
+			raw[i] = r
+		}
+	})
+
+	defer s.Pause(func() {
+		for _, r := range raw {
+			assert.NoError(r.(io.ReadCloser).Close())
+		}
+	})
+
+	blob := types.NewBlob(io.MultiReader(raw...))
+	fmt.Fprintf(s.W, "csv/raw is %s\n", humanize.Bytes(blob.Len()))
+
+	ds := dataset.NewDataset(s.Database, "csv/raw")
+	_, err := ds.CommitValue(blob)
+	assert.NoError(err)
+}
+
+func (s *perfSuite) Test02ImportSfCrimeCSVFromBlob() {
+	assert := s.NewAssert()
+
+	var importCmd *exec.Cmd
+
+	s.Pause(func() {
+		// Trick the temp file logic into creating a unique path for the csv-import binary.
+		f := s.TempFile()
+		f.Close()
+		out := f.Name()
+		os.Remove(out)
+
+		// Build csv-import into that temporary file.
```

Comment:
This sort of thing makes me wonder if we should add a notion of Suite setup. So that, for example, you could just build this binary once and then use it many times to import many different datasets.